megamek/src/megamek/common/EquipmentType.java

```diff
@@ -682,7 +679,6 @@ public boolean isNextTurnModeSwitch(String mode) {
      * @see #hasModes()
      */
     public EquipmentMode getMode(int modeNum) {
-        assert ((modes != null) && (modeNum >= 0) && (modeNum < modes.size())) : "Invalid Mode";
```

Comment:
I feel like these two methods are used enough and critical enough that "manual" validation is appropriate (obviously the asserts don't work, but...)

Reply:
Given it's worked without the checks thus far I think it's fine to skip... and is a definite slowdown to add that level of check to.